app/services/ApplicationService.scala

```diff
@@ -0,0 +1,16 @@
+package services
+
+import connectors.LibraryConnector
+import models.{DataModel, GoogleBook}
+
+import java.awt.print.Book
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ApplicationService @Inject()(connector: LibraryConnector) {
+
+  def getGoogleBook(urlOverride: Option[String] = None, search: String, term: String)(implicit ec: ExecutionContext): Future[DataModel] =
```

Comment:
Might need to change either this or the connector such that the types line up and the code compiles